js_modules/dagster-ui/packages/ui-core/src/metadata/OverflowingTagCollection.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  Button,
+  ButtonLink,
+  Colors,
+  Dialog,
+  DialogBody,
+  DialogFooter,
+  Tag,
+} from '@dagster-io/ui-components';
+import {useEffect, useRef, useState} from 'react';
+import styled from 'styled-components';
+
+import {DefinitionTag} from '../graphql/types';
+import {RunTags} from '../runs/RunTags';
+import {buildTagString} from '../ui/tagAsString';
+
+const SHOW_EXTRA_TAGS_WIDTH = 20;
+
+const getOverflowingTag = (target: HTMLDivElement | null) => {
+  if (!target || target.offsetWidth >= target.scrollWidth) {
+    return false;
+  }
+  // find the first tag that is overflowing
+  for (let i = 0; i < target.children.length; i++) {
+    const child = target.children[i];
+    if (
+      child &&
+      child?.getBoundingClientRect()?.right >
+        target.getBoundingClientRect().right - SHOW_EXTRA_TAGS_WIDTH
+    ) {
+      console.log(
+        child,
+        child?.getBoundingClientRect()?.right,
+        target.getBoundingClientRect().right,
+      );
+      return i;
+    }
+  }
+  return target.children.length;
+};
+
+type OverflowingTagCollectionProps = {
+  tags: DefinitionTag[];
+};
+
+export const OverflowingTagCollection = ({tags}: OverflowingTagCollectionProps) => {
+  const [showColumnTags, setShowColumnTags] = useState(false);
+  const [overflowing, setOverflowing] = useState<number | false>(false);
+  const container = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const tag = getOverflowingTag(container.current);
+    if (tag) {
+      setOverflowing(tag);
+    }
+  }, [overflowing]);
+
+  const tagsToRender = overflowing !== false ? tags.slice(0, overflowing) : tags;
+  const hiddenTagNumber = tags.length - tagsToRender.length;
+
+  const openColumnTagDialog = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    e.preventDefault();
+    setShowColumnTags(true);
+  };
```

Comment:
I don't think you should have to `stopPropagation` or `preventDefault` on this click, unless your component is nested in something that is expecting clicks.